scripts/firedrake-install

```diff
@@ -1577,6 +1577,7 @@ if mode == "install" or not args.update_script:
                              "cmake",
                              "gfortran",
                              "git",
+                             "pkg-config", # for p4est
```

Comment:
Can you try with pkgconf?

> pkgconf is a newer, actively maintained implementation of pkg-config that supports more aspects of the pkg-config file specification and provides a library interface that applications can use to incorporate intelligent handling of pkg-config files into themselves

https://fedoraproject.org/wiki/Changes/pkgconf_as_system_pkg-config_implementation

I'm slightly surprised this isn't included in `build-essential`.

Could you also look up the corresponding package to install on MacOS?

Reply:
> Can you try with pkgconf?

Package `pkgconf` works well on wsl-ubuntu


> I'm slightly surprised this isn't included in build-essential.

The package is not in build-essential. The following command show the packages included in build-essential
```bash
cat /usr/share/build-essential/list
```


> Could you also look up the corresponding package to install on MacOS?

Sorry, I don't have a Mac laptop.


I found an issue with respect to `apt-check` in the install script: when a package is removed, `apt-check` cannot give the correct result as `dpkg-query` returns without error. See the following output after `pkgconf` was removed.
```bash
yzz@AMA-PC-RA18:~$ dpkg-query -s pkgconf
Package: pkgconf
Status: deinstall ok config-files
Priority: optional
Section: devel
Installed-Size: 126
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Multi-Arch: foreign
Version: 1.8.0-1
Config-Version: 1.8.0-1
Provides: pkg-config (= 0.29-1)
Depends: libdpkg-perl, dpkg-dev, perl:any, libc6 (>= 2.34), libpkgconf3 (>= 1.8.0-1)
Breaks: pkg-config (>= 0.29-1)
Conffiles:
 /etc/dpkg/dpkg.cfg.d/pkgconf-hook-config a1673bd861645834495cb42519ddf648
Description: manage compile and link flags for libraries
 pkgconf is a replacement for pkg-config, a system for managing library
 compile and link flags that works with automake and autoconf.
 .
 pkgconf does not bundle any third-party libraries or depend on any
 third-party libraries.
Original-Maintainer: Andrej Shadura <andrewsh@debian.org>
Homepage: http://pkgconf.org/
```